File: javaSample/src/main/java/jp/ac/ritsumei/rcc/sample/exception/SampleException.java
package jp.ac.ritsumei.rcc.sample.exception;

import jp.ac.ritsumei.rcc.sample.model.ExceptionResponse;

public class SampleException extends Exception {
    protected ExceptionResponse exceptionResponse;

    public SampleException() {
        super();
        exceptionResponse = new ExceptionResponse();
    }

    public SampleException(String message) {
        super(message);
        exceptionResponse = new ExceptionResponse();
        exceptionResponse.setMessage(message);
    }
}

File: kotlinSample/src/main/kotlin/jp/ac/ritsumei/rcc/sample/exception/SampleException.kt
package jp.ac.ritsumei.rcc.sample.exception

import jp.ac.ritsumei.rcc.sample.model.ExceptionResponse

class SampleException : Exception {
    protected val exceptionResponse: ExceptionResponse

    constructor() : super() {
        exceptionResponse = ExceptionResponse()
    }

    constructor(message: String) : super(message) {
        exceptionResponse = ExceptionResponse(message)
    }
}